core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/EntityStatisticsCacheMonitor.scala

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import akka.actor._
+import akka.pattern._
+import cats.effect.{ContextShift, IO}
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource
+import org.broadinstitute.dsde.rawls.monitor.EntityStatisticsCacheMonitor.{EntityStatisticsCacheMessage, HandleBacklog, Sweep}
+import slick.dbio.DBIO
+
+import java.sql.Timestamp
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+object EntityStatisticsCacheMonitor {
+  def props(datasource: SlickDataSource, limit: Int)(implicit executionContext: ExecutionContext, cs: ContextShift[IO]): Props = {
+    Props(new EntityStatisticsCacheMonitorActor(datasource, limit))
+  }
+
+  sealed trait EntityStatisticsCacheMessage
+  case object Sweep extends EntityStatisticsCacheMessage
+  case object HandleBacklog extends EntityStatisticsCacheMessage
+}
+
+class EntityStatisticsCacheMonitorActor(val dataSource: SlickDataSource, val limit: Int)(implicit val executionContext: ExecutionContext, val cs: ContextShift[IO]) extends Actor with EntityStatisticsCacheMonitor with LazyLogging {
+
+  override def preStart(): Unit = {
+    super.preStart()
+    self ! Sweep
+  }
+
+  override def receive = {
+    case Sweep => sweep() pipeTo self
+    case HandleBacklog => handleBacklog() pipeTo self
+  }
+
+}
+
+trait EntityStatisticsCacheMonitor extends LazyLogging {
+
+  implicit val executionContext: ExecutionContext
+  implicit val cs: ContextShift[IO]
+  val dataSource: SlickDataSource
+  val limit: Int
+
+  def sweep() = {
+    val checkFuture = for {
+      recordsToUpdate <- dataSource.inTransaction { dataAccess =>
+        dataAccess.workspaceQuery.listOutdatedEntityCaches(limit)
+      }
+
+      updateResults <- recordsToUpdate.toList.traverse { case (workspaceId, lastModified) =>
+        IO.fromFuture(IO(updateStatisticsCache(workspaceId, lastModified))).map { _ =>
+          (workspaceId, lastModified.getTime, System.currentTimeMillis())
+        }
+      }.unsafeToFuture()
+    } yield {
+      updateResults.foreach { case (workspaceId, lastModified, cacheUpdated) =>
+        logger.info(s"Updated workspace $workspaceId. Cache was ${cacheUpdated - lastModified} millis out of date")
+      }
+      logger.info(s"Sweep complete. Updated entity cache for ${recordsToUpdate.length} workspace(s), with the limit set to $limit")
+      if(recordsToUpdate.length < limit) HandleBacklog
+      else Sweep
+    }
```

Comment:
I think this still needs some kind of poll interval or delay. While there are workspaces to update - including the backlog  -  it's good to work as fast as possible (I wonder if we want a few-millisecond delay just so we don't overwhelm the db). But ... once we've caught up on all workspaces, we should not be sweeping as fast as possible; there should be a delay once we've caught up.

Reply:
I've added a concept of a delayed sweep if we're caught up